### PR TITLE
Monocular depth estimation - Keras 3 Migration (Only Tensorflow Backend)

### PR DESCRIPTION
**Pull request by chunduriv**
This PR changes the Monocular depth estimation example to keras 3.0 (Tensorflow backend only), as requested in https://github.com/keras-team/keras-cv/issues/2211

Please review the attached [gist](https://colab.sandbox.google.com/gist/chunduriv/994b36a97985e44d9573436b987993f9/tf_depth_estimation.ipynb).

### CODE REVIEW

**Review by divyashreepathihalli — CHANGES_REQUESTED** [review]
Thanks for the PR @chunduriv! left a comment regarding implementing the image gradients.

File: examples/vision/depth_estimation.py [hunk 1]
```diff
@@ -105,7 +106,7 @@
 """
 
 
-class DataGenerator(tf.keras.utils.Sequence):
+class DataGenerator(keras.utils.Sequence):
```
Comment (fchollet) [inline examples/vision/depth_estimation.py:109]: You can use the up-to-date `PyDataset` class here

File: examples/vision/depth_estimation.py [hunk 2]
```diff
@@ -28,14 +28,15 @@
 import sys
 
 import tensorflow as tf
```
Comment (fchollet) [inline examples/vision/depth_estimation.py:30]: Since this example is TF-only, please add a line to set the backend to TF via `os.environ["KERAS_BACKEND"] = "tensorflow"`

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/vision/depth_estimation.py (hunk 1 above)
Comment (chunduriv) [reply examples/vision/depth_estimation.py]: Modified to PyDataset.

File: examples/vision/depth_estimation.py (hunk 2 above)
Comment (chunduriv) [reply examples/vision/depth_estimation.py:33]: Updated to Tensorflow Backend.

**Review by fchollet — APPROVED** [review]
LGTM, thanks!
